### ISSUE TITLE
when to close tickets, tracking fix verification status

### ISSUE THREAD

**Issue by Dieterbe** [raintank/meta#4]
@torkelo mentioned this the other day but we didn't actually commit to anything.
the idea of tracking tickets through the bugfix->deploy->verify->done cycle.
i know he wants to do lots of changes to our PM, possibly changing boards, but that's taking a while.
i don't see why we can't get started using this method with our current setup. there's only 2 tiny things that need to happen:
* add a "to verify" column to codetree between "in progress" and "done". I just did this.
* now simply don't close any tickets merely after pushing a bugfix to git. rather move it to the "to verify" column. once the thing is live, we actually make sure it works in production, and only then do we close the ticket/move to done column.

i'm deliberately keeping things very simple here. just 1 extra column. no extra "to-deploy-dev", "to-deploy-prod" columns etc.

this does come with a little bit of extra overhead which i know is not fun. but sometimes fixes don't actually seem to work, https://github.com/raintank/ops/issues/171 for example. and also external people are using our tickets to know when issues are fixed "for them" (i.e. live in prod), they don't care as much about it being in git or not.
and finally, right now we have a mixed approach. some tickets are closed when fix is in git, for others we wait when it's live or verified.  we should be more uniform.

@woodsaj @torkelo @ctdk @nopzor1200 do you think this is a good approach? or is there a better mechanism through which we can create higher reliability with less overhead?
one thing that i would miss is the ability to put `fix: #num` in a commit and have github auto-close the ticket on push to master.  i don't think we can make it reassign labels on push or anything like that.
yet another alternative would be create a new ticket or separate state somewhere to track whether the fix has been verified or not. we could then load in a badge in the original ticket, but it wouldn't inform people about a change.

so it seems like the extra verification step and keeping the ticket open longer is the least sucky solution that i can see.

**Comment by torkelo** [comment]
We need github labels for staging, prod, waiting-fo-deploy-to-prod , waiting-for-deploy-to-staging etc or something like that

**Comment by woodsaj** [comment]
I really feel that once fixes are committed to master, the issue should be closed.  Open issues are for outstanding development work that has not been completed.  With the large number of repositories we have, it is hard enough to keep track of what needs to be done without complicating matters by keeping issues open after they have been fixed.  GitHub issues are for managing development work, they are not the right place to be tracking deployment to the raintankSaaS platform.

**Comment by woodsaj** [comment]
Also, on the idea of verifying commits after merging to master is the wrong approach.  Code should already be tested and verified before it is merged to master.  

Deploying to dev-portal is not a way to test that your code works, it is a final safety net to catch any odd issues that only present themselves at production workloads.

**Comment by torkelo** [comment]
I agree that we can close issues when merged to master, but we can still use labels or milestones to track where it is in a workflow.

**Comment by ctdk** [comment]
I'm fine with closing tickets after they're committed to master, so long as we be consistent about it. My usual workflow has been to wait to close it out until it gets deployed to production, but that can give an inaccurate impression of what its status is.

**Comment by Dieterbe** [comment]
My main concern was external people following a ticket who are affected by an issue. As long as they can get a notification that the code pertaining to the ticket has been deployed that should be good. Ie we should come up with a workflow where we can close tickets when code is fixed and still update the ticket somehow with label assign/milestone change /... in a way that sends a notification to interested parties.    I also think there exists a category of issues that require follow up or are of critical nature and should (also) be verified once deployed or can only be verified in prod. And it would be nice to have a process that supports that elegantly as well.

**Comment by Dieterbe** [comment]
Also @mattttt any thoughts.

**Comment by mattttt** [comment]
I dont have any thoughts, this issue seems stale but it's unanswered by me.

**Comment by bergquist** [comment]
@mattttt the unanswered filter is kind of broken. Don't you think?
